util/profile_generator.py

```diff
@@ -0,0 +1,82 @@
+import os
+import json
+import sys
+import random
+from torrentool.api import Torrent
+
+from datetime import timedelta, datetime
+
+# dates for random uplaod time generation
+START_DATE = datetime.strptime("1/1/2023", "%d/%m/%Y")
+END_DATE = datetime.strptime("25/4/2023", "%d/%m/%Y")
+
+
+def random_date(start, end):
+    """
+    This function will return a random datetime between two datetime
+    objects.
+    """
+    delta = end - start
+    int_delta = (delta.days * 24 * 60 * 60) + delta.seconds
+    random_second = random.randrange(int_delta)
+    return start + timedelta(seconds=random_second)
+
+
+def get_random_torrents() -> list[Torrent]:
+    torrent_list = os.listdir("torrents/")
+
+    # randomly select at most half of the available torrents
+    num_torrents = random.randint(1, len(torrent_list) // 2)
+
+    selected_torrents = random.sample(torrent_list, num_torrents)
+
+    to_return = [Torrent.from_file("torrents/" + i) for i in selected_torrents]
+
+    return to_return
+
+
+def main():
+    try:
+        num_nodes = int(sys.argv[1])
+    except:
+        print("Usage: python3 profile_generator.py <num_nodes>")
+        exit(1)
+
+    output = {"nodes": []}
+
+    for _ in range(num_nodes):
+        node = {"profiles": []}
+        torrents = get_random_torrents()
+
+        for t in torrents:
```

Comment:
in #62 i iterate over all the videos in a torrent, and send a message per video to each peer, i don't think we need to change much for the dummy nodes as we just want to test the functionality but i wanted to discuss this regardless